crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs

```diff
@@ -84,30 +88,41 @@ impl OperatorInfoServiceInMemory {
 
         let subcription_new_operator_registration_stream = provider.subscribe_logs(&filter).await?;
         let mut stream = subcription_new_operator_registration_stream.into_stream();
-        while let Some(log) = stream.next().await {
-            let data = log
-                .log_decode::<BLSApkRegistry::NewPubkeyRegistration>()
-                .ok();
-
-            if let Some(new_pub_key_event) = data {
-                let event_data = new_pub_key_event.data();
-                let operator_pub_key = OperatorPubKeys {
-                    g1_pub_key: G1Point {
-                        X: event_data.pubkeyG1.X,
-                        Y: event_data.pubkeyG1.Y,
-                    },
-                    g2_pub_key: G2Point {
-                        X: event_data.pubkeyG2.X,
-                        Y: event_data.pubkeyG2.Y,
-                    },
-                };
-                // send message
-                let _ = self.pub_keys.send(OperatorsInfoMessage::InsertOperatorInfo(
-                    event_data.operator,
-                    operator_pub_key,
-                ));
+
+        let shutdown_rx = shutdown_rx.clone();
+        let pub_keys = self.pub_keys.clone();
+        tokio::spawn(async move {
+            while let Some(log) = stream.next().await {
+                if shutdown_rx.has_changed().unwrap_or(false) {
+                    println!("shutdown ");
+                    break;
+                }
+
+                let data = log
+                    .log_decode::<BLSApkRegistry::NewPubkeyRegistration>()
+                    .ok();
+
+                if let Some(new_pub_key_event) = data {
+                    let event_data = new_pub_key_event.data();
+                    let operator_pub_key = OperatorPubKeys {
+                        g1_pub_key: G1Point {
+                            X: event_data.pubkeyG1.X,
+                            Y: event_data.pubkeyG1.Y,
+                        },
+                        g2_pub_key: G2Point {
+                            X: event_data.pubkeyG2.X,
+                            Y: event_data.pubkeyG2.Y,
+                        },
+                    };
+                    // Send message
+                    let _ = pub_keys.send(OperatorsInfoMessage::InsertOperatorInfo(
```

Comment:
should we log here?

Reply:
Added a TODO to log after this gets merged, will do in #62

Reply:
added loggiing